src/app/forms/custom-sections/body/body.component.ts

```diff
@@ -153,4 +151,12 @@ export class BodyComponent implements OnInit, OnChanges, OnDestroy {
       this.optionsService.loadOptions(ReferenceDataResourceType.TrlMake);
     }
   }
+
+  updateHgvVehicleBodyType() {
+    if (this.techRecord.techRecord_vehicleType === 'hgv' && this.techRecord.techRecord_vehicleConfiguration === 'articulated') {
```

Comment:
I think you already know the vehicle is hgv at this point from the parent calling, this is an interesting one. Again this looks like it will end up being a chain of conditional checks on vehicle configuration.

since really you're just passing it through arnt you actually just checking that its a valid configuration for a hgv and then you can handle anything thats not valid a specific way ?